drift/lib/src/runtime/manager/manager.dart

```diff
@@ -463,10 +705,19 @@ abstract class BaseTableManager<
   /// The [distinct] parameter (enabled by default) controls whether to generate
   /// a `SELECT DISTINCT` query, removing duplicates from the result.
   @override
-  Future<$Dataclass?> getSingleOrNull({bool distinct = true}) => $state
-      .copyWith(distinct: distinct)
-      .buildSelectStatement()
-      .getSingleOrNull();
+  Future<$ActiveDataclass?> getSingleOrNull({bool distinct = true}) async {
+    final list = await get(distinct: distinct);
+    final iterator = list.iterator;
+
+    if (!iterator.moveNext()) {
+      return null;
+    }
+    final element = iterator.current;
+    if (iterator.moveNext()) {
+      throw StateError('Expected exactly one result, but found more than one!');
+    }
+    return element;
```

Comment:
~I think we should mention the changed behavior of throwing if there are multiple rows~ (nevermind, updated to be consistent with the rest of drift). Instead of using the iterator manually we can also do

```dart
if (list.isNotEmpty) {
  return list.single;
} else {
  return null;
}
```

Reply:
I copy pasted this from the original implementation.

Should I update that too?

Reply:
Hm it looks like we have multiple copies of that around then, I've checked `singleElementsOrNull` which uses the `isNotEmpty` check. It might be worth extracting them all into a common method then.